generator/booking_quality_check/booking_quality_check.cpp

```diff
@@ -0,0 +1,109 @@
+#include "std/iostream.hpp"
+
+#include "generator/booking_dataset.hpp"
+#include "generator/osm_source.hpp"
+
+#include "geometry/distance_on_sphere.hpp"
+
+#include "3party/gflags/src/gflags/gflags.h"
+
+#include "std/fstream.hpp"
+
+DEFINE_bool(generate_classif, false, "Generate classificator.");
+
+DEFINE_string(osm_file_name, "", "Input .o5m file");
+DEFINE_string(booking_data, "", "Path to booking data in .tsv format");
+DEFINE_string(sample_data, "", "Sample output path");
+DEFINE_uint64(selection_size, 1000, "Selection size");
+
+using namespace generator;
+
+ostream & operator<<(ostream & s, OsmElement const & e)
+{
+  for (auto const & tag : e.Tags())
+  {
+    auto t = tag;
+    replace(t.key.begin(), t.key.end(), '\n', ' ');
+    replace(t.value.begin(), t.value.end(), '\n', ' ');
+    s << t.key << "=" << t.value << "\t";
+  }
+  return s;
+}
+
+int main(int argc, char * argv[])
+{
+  google::SetUsageMessage(
+      "Takes OSM XML data from stdin and creates"
+      " data and index files in several passes.");
+  google::ParseCommandLineFlags(&argc, &argv, true);
+
+  LOG_SHORT(LINFO, ("Booking data:", FLAGS_booking_data));
+
+  BookingDataset bookingDataset(FLAGS_booking_data);
+
+  vector<OsmElement> elements;
+  auto const counterAction = [&](OsmElement * e)
+  {
+    if (bookingDataset.TourismFilter(*e))
+      elements.emplace_back(*e);
+  };
+
+  LOG_SHORT(LINFO, ("OSM data:", FLAGS_osm_file_name));
+  {
+    SourceReader reader =
+        FLAGS_osm_file_name.empty() ? SourceReader() : SourceReader(FLAGS_osm_file_name);
+    ProcessOsmElementsFromO5M(reader, counterAction);
+  }
+  LOG_SHORT(LINFO, ("Tourism elements:", elements.size()));
+
+  vector<size_t> elementIndexes(elements.size());
+  size_t counter = 0;
+  for (auto & e : elementIndexes)
+    e = counter++;
+
+  random_shuffle(elementIndexes.begin(), elementIndexes.end());
```

Comment:
random_shuffle is deprecated. Also, please, add a flag for the seed, and use it here.